### ISSUE TITLE
support batching when fitting shallow classifiers on embeddings

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#1141]
Currently, we implement quick_fit for fitting classifiers without mini-batching. This is good for small training datasets, but we should implement mini-batching (1) for larger datasets and (2) because it enables the stochasticity of stochastic gradient descent. This will be a very simple and classic pytorch training loop. It probably makes sense to replace quick_fit with a single fit method with optional batching.
